### PR TITLE
refactor(logger): Update logger internals

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

- Switch from `shortid` to `nano-id`
- Restrict types

## Context:

- Ref: #7154

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: tsconfig.strict.json [hunk 1]
```diff
@@ -46,6 +46,12 @@
     "./lib/globals.d.ts",
     "./lib/logger/__mocks__/index.ts",
     "./lib/logger/cmd-serializer.ts",
+    // "./lib/logger/config-serializer.ts",
+    "./lib/logger/err-serializer.ts",
+    // "./lib/logger/index.ts",
```
Comment (zharinov) [inline tsconfig.strict.json:51]: Ready, but fails via upper-level dependency

**Comment by zharinov** [comment]
We may want to split changes, though maybe it's okay to merge this one

File: lib/logger/config-serializer.ts [hunk 2]
```diff
@@ -14,15 +14,17 @@ export default function configSerializer(
...
-    if (val && templateFields.includes(this.key)) {
-      this.update('[Template]');
-    }
-    if (val && contentFields.includes(this.key)) {
-      this.update('[content]');
-    }
-    // istanbul ignore if
-    if (val && arrayFields.includes(this.key)) {
-      this.update('[Array]');
+    if (this.key && val) {
```
Comment (viceice) [inline lib/logger/config-serializer.ts:17]: Why this change? Should we add a `this: xxx` type to function declaration?

File: lib/logger/index.ts [hunk 3]
```diff
@@ -1,15 +1,15 @@
 import is from '@sindresorhus/is';
 import * as bunyan from 'bunyan';
-import * as shortid from 'shortid';
+import { nanoid } from 'nanoid';
```
Comment (viceice) [inline lib/logger/index.ts:3]: What's the benefit?

File: lib/logger/utils.ts [hunk 4]
```diff
@@ -113,9 +114,7 @@ export function sanitizeValue(_value: unknown, seen = new WeakMap()): any {
     value = prepareError(value);
   }
 
-  const valueType = typeof value;
-
-  if (value && valueType !== 'function' && valueType === 'object') {
+  if (value && typeof value !== 'function' && typeof value === 'object') {
```
Comment (viceice) [inline lib/logger/utils.ts:117]: Shouldn't typescript already be able to track that?

@JamieMagee ideas?

File: lib/logger/index.ts (hunk 3 above)
Comment (zharinov) [reply lib/logger/index.ts:3]: Shortid is deprecated, and its typings seemed to be missing

File: lib/logger/config-serializer.ts (hunk 2 above)
Comment (zharinov) [reply lib/logger/config-serializer.ts:17]: Don't know, its definition led me to traverse inner interface with `key: string | undefined`

File: lib/logger/index.ts (hunk 3 above)
Comment (viceice) [reply lib/logger/index.ts:3]: Ok, i saw we already have nanoid as transitive dep.

File: tsconfig.strict.json (hunk 1 above)
Comment (zharinov) [reply tsconfig.strict.json:51]: Convered this PR to draft until resolution of top-level dependency strictness

File: lib/logger/config-serializer.ts (hunk 2 above)
Comment (zharinov) [reply lib/logger/config-serializer.ts:17]: To be more concrete, `this` is already inferred `traverse.TraverseContext` interface with `key: string | undefined` field. We have to make sure `key` is string before searching it in `string[]` arrays.

File: lib/logger/utils.ts (hunk 4 above)
Comment (zharinov) [reply lib/logger/utils.ts:117]: Fixed here: https://github.com/renovatebot/renovate/pull/12859

**Review by viceice — APPROVED** [review]
